lib/_stream_writable.js

```diff
@@ -108,6 +108,31 @@ function WritableState(options, stream) {
 
   // True if the error was already emitted and should not be thrown again
   this.errorEmitted = false;
+
+  // count buffered requests
+  this.bufferedRequests = 0;
```

Comment:
Nit: I'd lean towards naming this attribute something like `bufferedRequestLength` or `bufferedRequestCount` just to make it clear to readers that we're tracking a number.